lib/tasks/one_time/2017_Q4/delete_invalid_addresses.rake

```diff
@@ -0,0 +1,35 @@
+# PREREQUISITE MIGRATIONS:
+# None
+
+# PREREQUISITE RAKE TASKS:
+# None
+
+namespace :shf do
+  namespace :addresses do
+
+    desc "Delete invalid addresses in DB"
+    task delete_invalid: :environment do
+
+      LOG_FILE = 'log/delete_invalid_addresses'
+
+      ActivityLogger.open(LOG_FILE, 'SHF_TASK', 'Delete addresses') do |log|
+
+        log.record('info', "Checking validity of #{Address.count} addresses.")
+
+        deletes = 0
+
+        Address.all.each do |address|
+          next if address.valid?
+
+          log.record('info', "Deleting Address ID: #{address.id}")
+
```

Comment:
can we also log `Address was: #{address.inspect}`  just in case there was some little bits of info in there.